components/pedigree/ui/src/main/resources/PhenoTips/PedigreeMacros.xml

```diff
@@ -117,4 +198,436 @@
+
+    if (pedigreeTrigger) {
+       pedigreeTrigger.invoke('observe', 'click', function(event) {
+        if ($('inline')) { //if edit mode
+          event.stop();
+          document.observe("xwiki:document:saved", openPedigreeEditor);
+          document.fire("xwiki:actions:save", {"continue" : true, "form" : $('inline')});
+        } else {
+          openPedigreeEditor();
+        }
+      });
+    }
+
+    var familyInterfaceDoc = new XWiki.Document('FamilyPedigreeInterface', 'PhenoTips');
+    var baseUrl = familyInterfaceDoc.getURL('get', 'rand='+ Math.random());
+
+    // Delete Pedigree functionality for Patient Form
+    if (XWiki.currentDocument.space == 'data') {
+
+      var pedigreeDelete = $("pedigree-delete");
+      if (pedigreeDelete) {
+        pedigreeDelete.observe("click", function(event) {
+          event.stop();
+          var familyID = $('familyIdentifier').value;
+          var fullURL = baseUrl + '&amp;action=familyinfo&amp;document_id=' + familyID;
+          new Ajax.Request(fullURL, {
+            method: "POST",
+            onSuccess: onFamilyDataReadyBeforeDeletion.bind(this)
+          });
+        });
+      }
+
+      var onFamilyDataReadyBeforeDeletion = function (response) {
+        if (response &amp;&amp; response.responseJSON) {
+
+          var familyMembers = response.responseJSON.family.hasOwnProperty("familyMembers") ? response.responseJSON.family.familyMembers: [];
+          var familyId = response.responseJSON.family.id;
+          var familyExternalId = response.responseJSON.family.externalId;
+
+          var dialogParameters = { showCancelButton: false };
+          dialogParameters.confirmationText = "$services.localization.render('phenotips.patientRecord.pedigree.deleteWarning')";
+
+          var constructDialog = function(dialogParameters, canDeleteFamily) {
+            var confirmationDialog = new XWiki.widgets.ConfirmationBox(
+              {
+                onYes : function() {
+                  var parameters = {"action": "removepatient", "patient_id": XWiki.currentDocument.page};
+                  if ($('inline')) { //if edit mode, save before proceed
+                    document.fire("xwiki:actions:save", {"continue" : true, "form" : $('inline'), 'confirmed' : true});
+                    document.observe("xwiki:document:saved", function () {
+                      sendDeleteRequest(parameters);
+                    });
+                  } else {
+                    sendDeleteRequest(parameters);
+                  }
+                },
+                onNo : function() { }
+              },
+              dialogParameters
+            );
+
+            // if user has rights to delete family, add one more button
+            if (canDeleteFamily) {
+              var deleteFamilyButton = new Element('input', {type: 'button', 'value': '$services.localization.render('phenotips.patientRecord.pedigree.deleteFamily')', 'class' : 'button  delete-family', 'id': 'delete_family'});
```

Comment:
should be `"$services.localization.render('phenotips.patientRecord.pedigree.deleteFamily')"`